app/lib/applets/lessons/student/lessons_student_view.dart

```diff
@@ -15,65 +16,126 @@ class LessonsStudentView extends StatefulWidget {
-
+class _LessonsStudentViewState extends State<LessonsStudentView>
+    with TickerProviderStateMixin {
   Widget noDataScreen(context) => Center(
-    child: Column(
-      mainAxisAlignment: MainAxisAlignment.center,
-      crossAxisAlignment: CrossAxisAlignment.center,
-      children: [
-        const Icon(
-          Icons.search,
-          size: 60,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            const Icon(
+              Icons.search,
+              size: 60,
+            ),
+            Text(AppLocalizations.of(context)!.noCoursesFound)
+          ],
         ),
-        Text(AppLocalizations.of(context)!.noCoursesFound)
-      ],
-    ),
-  );
+      );
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: widget.openDrawerCb != null ? AppBar(
-        title: Text(lessonsDefinition.label(context)),
-        leading: IconButton(
-          icon: const Icon(Icons.menu),
-          onPressed: () => widget.openDrawerCb!(),
-        ),
-      ) : null,
       body: CombinedAppletBuilder<Lessons>(
-      parser: sph!.parser.lessonsStudentParser,
+        parser: sph!.parser.lessonsStudentParser,
         phpUrl: lessonsDefinition.appletPhpUrl,
         settingsDefaults: lessonsDefinition.settingsDefaults,
         accountType: sph!.session.accountType,
-        builder: (context, lessons, accountType, settings, updateSetting, refresh) {
-          if (lessons.isEmpty) return noDataScreen(context);
-          Lessons attendanceLessons = lessons.where((element) => element.attendances != null).toList();
+        builder:
+            (context, lessons, accountType, settings, updateSetting, refresh) {
+          Lessons? attendanceLessons;
+
+          if (settings['showHomework'] == 'true') {
+            lessons = lessons
+                .where((element) => element.currentEntry?.homework != null)
+                .toList();
+
+            lessons.sort((a, b) {
+              if (a.currentEntry!.homework!.homeWorkDone ==
+                  b.currentEntry?.homework?.homeWorkDone) {
+                if (a.currentEntry?.topicDate != null &&
+                    b.currentEntry?.topicDate != null) {
+                  return a.currentEntry!.topicDate!
+                      .compareTo(b.currentEntry!.topicDate!);
+                } else {
+                  return a.currentEntry?.topicDate == null ? 1 : -1;
+                }
+              }
+              return (a.currentEntry?.homework?.homeWorkDone ?? false) ? 1 : -1;
+            });
+          } else {
+            attendanceLessons = lessons
+                .where((element) => element.attendances != null)
+                .toList();
+          }
 
           return Scaffold(
+            appBar: widget.openDrawerCb != null
+                ? AppBar(
```

Comment:
The app bar has to be located outside of the CombinedAppletBuilder, to ensure it is rendered and the user has the ability to open the sidebar while on a loading screen.